TicTacToe.sol

```diff
@@ -2,36 +2,9 @@
 pragma solidity ^0.8.0;
 
 contract TicTacToe {
-    constructor() {
-        createWinningCases();
-    }
-
-    WinningCase[] private winningCases;
-
-    struct WinningCase {
-        uint firstIndex;
-        uint secondIndex;
-        uint thirdIndex;
-    }
-
-    function createWinningCases() private {
-        WinningCase memory case1 = WinningCase(0, 1, 2);
-        WinningCase memory case2 = WinningCase(3, 4, 5);
-        WinningCase memory case3 = WinningCase(6, 7, 8);
-        WinningCase memory case4 = WinningCase(0, 3, 6);
-        WinningCase memory case5 = WinningCase(1, 4, 7);
-        WinningCase memory case6 = WinningCase(2, 5, 8);
-        WinningCase memory case7 = WinningCase(0, 4, 8);
-        WinningCase memory case8 = WinningCase(2, 4, 6);
-        winningCases.push(case1);
-        winningCases.push(case2);
-        winningCases.push(case3);
-        winningCases.push(case4);
-        winningCases.push(case5);
-        winningCases.push(case6);
-        winningCases.push(case7);
-        winningCases.push(case8);
-    }
+    int8 private constant EMPTY_MOVE = 0;
+    int8 private constant O_MOVE = 1;
+    int8 private constant X_MOVE = 2;
```

Comment:
Making `EMPTY_MOVE = 0` instead of `EMPTY_MOVE = 3` saves gas (and also solves a vulnerability) because Solidity will initialized each value to `0` when creating a new `board`, so you no longer have to do this:

```solidity
int8[9] private emptyBoard = [
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE,
        EMPTY_MOVE
    ];
```